tests/test_telemetry_interface.ts

```
@@ -0,0 +1,303 @@
+
+      const activeContext = telemetryClient.contextManager.active();
+      activeContext['getSpan'] = () => span;
+
+      try {
+        await queue.add('testJob', { foo: 'bar' });
+
+        expect.fail('Expected an error to be thrown');
+      } catch (error) {
+        span.recordException(error);
+
+        sinon.assert.calledOnce(recordExceptionSpy);
+        const [exception] = recordExceptionSpy.firstCall.args;
+        expect(exception?.message).to.equal('Simulated error');
+      } finally {
+        addStub.restore();
+      }
+    });
+  });
+
+  describe('Queue.addBulk', () => {
+    it('should correctly interact with telemetry when adding multiple jobs', async () => {
+      const jobs = [
+        { name: 'job1', data: { foo: 'bar' } },
+        { name: 'job2', data: { baz: 'qux' } },
+      ];
+
+      await queue.addBulk(jobs);
+
+      const activeContext = telemetryClient.contextManager.active();
+      const span = activeContext.getSpan?.() as MockSpan;
+
+      expect(span).to.be.an.instanceOf(MockSpan);
+      expect(span.name).to.equal(`${queueName} Queue.addBulk`);
+      expect(span.options?.kind).to.equal(SpanKind.PRODUCER);
+      expect(span.attributes[TelemetryAttributes.BulkNames]).to.deep.equal(
+        jobs.map(job => job.name),
+      );
+      expect(span.attributes[TelemetryAttributes.BulkCount]).to.equal(
+        jobs.length,
+      );
+    });
+
+    it('should correctly handle errors and record them in telemetry for addBulk', async () => {
+      const jobs = [
+        { name: 'job1', data: { foo: 'bar' } },
+        { name: 'job2', data: { baz: 'qux' } },
+      ];
+
+      const addBulkStub = sinon
+        .stub(queue.Job, 'createBulk')
+        .rejects(new Error('Simulated bulk error'));
+
+      const span = telemetryClient.trace
+        .getTracer('testtracer')
+        .startSpan('Queue.addBulk.error') as MockSpan;
+      const recordExceptionSpy = sinon.spy(span, 'recordException');
```

Same comment as above, in order for this test to be meaningful it should throw an exception from inside addBulk.